src/Core/Checkout/Cart/Calculator.php

```diff
@@ -103,20 +93,14 @@ private function calculatePrice(Cart $cart, LineItem $lineItem, CheckoutContext
 
         $definition = $lineItem->getPriceDefinition();
 
-        if ($definition instanceof AbsolutePriceDefinition) {
-            //reduce line items for provided filter
-            $prices = $this->filterLineItems($calculated, $definition->getFilter(), $context)
-                ->getPrices();
+        foreach ($this->calculators as $calculator) {
+            if ($calculator->supports($definition)) {
```

Comment:
First calculator is supported wins.
Is this expected or should there only be exact one supported calculator per `calculate()` run.
Maybe throw an LogikException if more than one Calculator supports given `$definition`.